### ISSUE TITLE
FirestoreImpl initialization needs to be delayed

### ISSUE THREAD

**Issue by mar9000** [quarkiverse/quarkus-google-cloud-services#90]
Hi, and thanks for this extension.

With:
  * quarkus 1.12.2.Final
  * google services firestore 0.5.0

I've tested the `FirestoreResource` on the [README.md](https://github.com/quarkiverse/quarkus-google-cloud-services/tree/master/firestore) build a sample project exactly as stated on the same file. It compile to native without problem.

Nevertheless adding these lines (note that the documentId is now generated):
```java
futures.add(persons.document().set(new Person(1L, "John", "Doe")));
futures.add(persons.document().set(new Person(2L, "Jane", "Doe")));
futures.add(persons.document().set(new Person(3L, "Charles", "Baudelaire")));
```
the native compilation fails with:
```
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of sun.security.provider.NativePRNG are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=sun.security.provider.NativePRNG.
Detailed message:
Trace: Object was reached by 
	reading field java.security.SecureRandom.secureRandomSpi of
		constant java.security.SecureRandom@701e41f6 reached by 
	scanning method com.google.cloud.firestore.FirestoreImpl.autoId(FirestoreImpl.java:84)
Call path from entry point to com.google.cloud.firestore.FirestoreImpl.autoId(): 
	at com.google.cloud.firestore.FirestoreImpl.autoId(FirestoreImpl.java:81)
	at com.google.cloud.firestore.CollectionReference.document(CollectionReference.java:103)
	at org.acme.resteasyjackson.FirestoreResource.firestore(FirestoreResource.java)
```
This property solve the problem(see also #67):
```properties
quarkus.native.additional-build-args=--initialize-at-run-time=com.google.cloud.firestore.FirestoreImpl
```
Issue #67 is closed so I decided to reopen a new one.

**Comment by loicmathieu** [comment]
@dzou do you think it's something that should be fixed upstream in the native support google cloud library or should I fixed this issue on the Quarkus side (it's easy on the extension side to delay the initialization).

**Comment by dzou** [comment]
Ah, thanks for the report.

@loicmathieu - Yes indeed, we will fix this on our end and I will prepare a release for you to use.

**Comment by loicmathieu** [comment]
@dzou thanks, I flagged the issue as upstream so we can easily track these kind of issue that should be fixed upstream.

**Comment by loicmathieu** [comment]
@dzou thanks for fixing it upstream.

But you was a little too hurry to close this issue as it's not yet available here ;)

I reopen the issue, we'll close it when the upstream fix arrives here (so when a new version of the google-cloud-graalvm-support library is released and integration here).

**Comment by dzou** [comment]
Oops, sorry, didn't mean to close it here! I submitted the fix upstream; will do a new release and then upgrade the version in here.

**Comment by loicmathieu** [comment]
Fixed via #96
